ci: move shellcheck in separate action

rst0git opened the pull request:
ShellCheck is one of the apt packages installed by default in the GitHub Action [virtual environment](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md).

Moving shellcheck in a separate action allows for the Makefile CI and ShellCheck actions to run in parallel.

rst0git commented:
cc: @mestery

mestery commented:
Great fix @rst0git! Note that I'm fixing the CI in #90, I'd like to merge that first. Also, can you make the build CI job dependent on the shellcheck one instead of having them run in parallel? The reason I ask is that shellcheck is fast, and if it fails, we don't want to run the much longer build job. Does that make sense?

rst0git commented:
> Note that I'm fixing the CI in https://github.com/ipdk-io/ipdk/pull/90, I'd like to merge that first.

Sure, no problem. I rebased my patch on your branch and converted this PR to draft. I will mark this PR as "ready for review" once https://github.com/ipdk-io/ipdk/pull/90 has been merged.

> can you make the build CI job dependent on the shellcheck one instead of having them run in parallel?

Yes, I moved shellsheck as separate job instead of workflow and added the necessary dependency.

mestery commented:
@rst0git I merged #90, so if you rebase and push we can merge this one.

rst0git commented:
> I merged #90, so if you rebase and push we can merge this one.

Thanks, I've rebased on the main branch updated this pull request.

mestery reviewed (approved):
Thanks for working on this!
